notebook/pipeline/_notebook_op.py

```diff
@@ -60,15 +61,23 @@ def __init__(self,
         self.cos_dependencies_archive = cos_dependencies_archive
         self.container_work_dir = "jupyter-work-dir"
         self.bootstrap_script_url = bootstrap_script_url
+        self.requirements_url = requirements_url
         self.pipeline_outputs = pipeline_outputs
         self.pipeline_inputs = pipeline_inputs
 
         if self.bootstrap_script_url is None:
             """ If bootstrap_script arg with URL not provided, use the one baked in here.
             """
 
-            self.bootstrap_script_url = 'https://raw.githubusercontent.com/elyra-ai/' \
-                                        'kfp-notebook/master/etc/docker-scripts/bootstrapper.py'
+            self.bootstrap_script_url = 'https://raw.githubusercontent.com/akchinSTC/' \
+                                        'kfp-notebook/boot-version/etc/docker-scripts/bootstrapper.py'
+
+        if self.requirements_url is None:
+            """ If bootstrap_script arg with URL not provided, use the one baked in here.
+            """
+
+            self.requirements_url = 'https://raw.githubusercontent.com/akchinSTC/' \
+                                    'kfp-notebook/boot-version/etc/requirements.txt'
```

Comment:
Hmm. Looking at the bootstrap_script_url block, that file was always pulled from `master`, and I'm assuming the same would occur here.  However, it seems like the _elyra version_ should be used here and there be elyra-specific release directories in kfp-notebook's master branch in which `requirements.txt` and `bootstrapper.py` are located for each corresponding elyra release - otherwise aren't we subject to older releases of elyra suddenly trying to use master-level packages and bootstrap software?

Reply:
Correct, we are pulling scripts from the master branch. Moving forward I think we should instead be pulling from a tagged release of `kfp-notebook` instead of master. Maybe jinja-fy the URLs and filling in the tagged versions prior to publishing to pypi. Then we can avoid having multiple version directories.

Reply:
Ok.  I guess I was still in the mode that the requirements.txt entries were related to Elyra's version, but, you're right, they're more related to kfp-notebook's version so using a templated URL sounds like a great idea.  Perhaps if you get back a 404 (because the version is malformed or doesn't exist), you could fall back to `master` as the branch (but log that fact, if possible).

Reply:
I guess you this is all one big command and you don't have the luxury of handling a 404. 
 You'd need to test the URL prior to setting the target in into the kwargs array.

Hmm, have we looked into encoding bootstrapper.py and requirements.txt into base64 strings, then using a decoder as the command instead of `curl`?  If you could do that, then you know exactly which "version" of those files is being used because its the current kfp-notebook that is running which produces them. 

You'd then need to ensure simple base64-string-to-file code is present on the image, but I would imagine that kind of code is something that could be release-independent and always pulled directly from master.  Or could that be something an init-container does - essentially responsible for getting bootstrapper.py and requirements.txt into position?  

Just thinking out loud.